### ISSUE TITLE
checkSEGsOverlapping throwing Uncaught RangeError for examples/displaySegmentation/index.html

### ISSUE THREAD

**Issue by gangaramstyle** [dcmjs-org/dcmjs#184]
After the patch in Segmentation_4X.js to properly compute getSegmentIndex as described in https://github.com/dcmjs-org/dcmjs/issues/177, there are still a few more small issues to get examples/displaySegmentation/index.html to work without errors. All of the fixes, including the getSegmentIndex patch, are described in https://github.com/gangaramstyle/dcmjs/commit/c8a5ba9. @igoroctaviano has created a commit to fix getSegmentIndex with https://github.com/dcmjs-org/dcmjs/pull/183 but after that is patched we'll likely see the error

Start offset -153792576 is outside the bounds of the buffer
  | _loop4 | @ | Segmentation_4X.js:643
-- | -- | -- | --
  | checkSEGsOverlapping | @ | Segmentation_4X.js:634
  | generateToolState$1 | @ | Segmentation_4X.js:355
  | generateToolState$2 | @ | Segmentation.js:66
  | parseSeg | @ | index.html:129
  | (anonymous) | @ | index.html:98
  | load (async) |   |  
  | getAndLoadSeg | @ | index.html:97
  | onclick | @ | index.html:25

I haven't recursed into how exactly to fix this underlying error here, but the error can be bypassed if Segmentation_4X.js generateToolState skipOverlapping is set to true. Unfortunately the client can't set skipOverlapping to true because the client calls generateToolState from Segmentation.js which then calls Segmentation_4X.js generateToolState without passing a parameter from the client to skipOverlapping.

Segmentation_4X.js
```
code for function generateToolState(
    imageIds,
    arrayBuffer,
    metadataProvider,
    skipOverlapping = false
) {
```

Segmentation.js (current)
```
function generateToolState(
    imageIds,
    arrayBuffer,
    metadataProvider,
    cornerstoneToolsVersion = 4
) {
    if (cornerstoneToolsVersion === 4) {
        return Segmentation_4X.generateToolState(
            imageIds,
            arrayBuffer,
            metadataProvider
        );
    }

    if (cornerstoneToolsVersion === 3) {
        return Segmentation_3X.generateToolState(
            imageIds,
            arrayBuffer,
            metadataProvider
        );
    }
...
```

Segmentation.js (proposed)
```
function generateToolState(
    imageIds,
    arrayBuffer,
    metadataProvider,
    **skipOverlapping = false,**
    cornerstoneToolsVersion = 4
) {
    if (cornerstoneToolsVersion === 4) {
        return Segmentation_4X.generateToolState(
            imageIds,
            arrayBuffer,
            metadataProvider,
            **skipOverlapping**
        );
    }

    if (cornerstoneToolsVersion === 3) {
        return Segmentation_3X.generateToolState(
            imageIds,
            arrayBuffer,
            metadataProvider,
            **skipOverlapping**
        );
    }
...
```
Once https://github.com/dcmjs-org/dcmjs/pull/183 is merged, I can push a new PR that makes the modification in Segmentation.js from https://github.com/gangaramstyle/dcmjs/commit/c8a5ba9.

**Comment by gangaramstyle** [comment]
@igoroctaviano @Punzo

**Comment by pieper** [comment]
I'm trying to keep up with the conversations, but just to be sure do you have this one @Punzo ?

**Comment by Punzo** [comment]
@pieper see https://github.com/dcmjs-org/dcmjs/pull/187

**Comment by pieper** [comment]
> > but just to be sure do you have this one @Punzo ?

> @pieper see #187

I'll take that as a yes  ; )
